src/elysium/sigmawalletv1.cpp

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) 2020 The Zcoin Core Developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "sigmawalletv1.h"
+
+#include "../libzerocoin/Zerocoin.h"
+#include "../sigma/openssl_context.h"
+#include "../wallet/wallet.h"
+
+namespace elysium {
+
+SigmaWalletV1::SigmaWalletV1()
+    : SigmaWallet(new SigmaWalletV1::Database())
+{
+}
+
+bool SigmaWalletV1::GeneratePublicKey(ECDSAPrivateKey const &priv, secp256k1_pubkey &out)
+{
+    return secp256k1_ec_pubkey_create(
+        OpenSSLContext::get_context(),
+        &out,
+        priv.data());
+}
+
+void SigmaWalletV1::GenerateSerial(secp256k1_pubkey const &pubkey, secp_primitives::Scalar &serial)
```

Comment:
Better to return a `secp_primitives::Scalar` directly so it is more easy to use.

Reply:
good idea